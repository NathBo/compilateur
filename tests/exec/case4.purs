module Main where

import Prelude
import Effect
import Effect.Console

data Q = A | B
data T = U Q | V

f :: T -> String
f V = ">>>V<<<<"
f (U A) = ">>>>U A<<<<"
f (U B) = ">>>>U B<<<<"

main :: Effect Unit
main = do log (f (U B))
          log (f (U A))
          log (f (V))

