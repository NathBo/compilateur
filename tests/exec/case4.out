>>>>U B<<<<
>>>>U A<<<<
>>>V<<<<
